src/Microsoft.DotNet.VersionTools/Automation/PullRequestCreator.cs

```diff
@@ -0,0 +1,201 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.DotNet.VersionTools.Automation.GitHubApi;
+using Microsoft.DotNet.VersionTools.Util;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Threading.Tasks;
+
+namespace Microsoft.DotNet.VersionTools.Automation
+{
+    public class PullRequestCreator
+    {
+        private GitHubAuth _auth;
+
+        public string BranchPrefix { get; set; } = "UpdateDependencies";
+
+        public GitHubBranch UpstreamBranch { get; }
+        public GitHubProject Origin { get; }
+
+        public string GitAuthorName { get; }
+
+        public PullRequestCreator(
+            GitHubAuth auth,
+            string projectName,
+            string upstreamOwner,
+            string upstreamBranch,
+            string gitAuthorName)
+            : this(
+                auth,
+                new GitHubProject(projectName, auth.User),
+                new GitHubBranch(upstreamBranch, new GitHubProject(projectName, upstreamOwner)),
+                gitAuthorName)
+        {
+        }
+
+        public PullRequestCreator(
+            GitHubAuth auth,
+            GitHubProject origin,
+            GitHubBranch upstreamBranch,
+            string gitAuthorName = null)
+        {
+            _auth = auth;
+
+            Origin = origin;
+            UpstreamBranch = upstreamBranch;
+
+            GitAuthorName = gitAuthorName ?? auth.User;
+        }
+
+        public async Task CreateOrUpdateAsync(
+            string commitMessage,
+            string title,
+            string description,
+            bool forceCreate = false)
+        {
+            using (var client = new GitHubClient(_auth))
+            {
+                var originBranch = new GitHubBranch(
+                    $"{BranchPrefix}{DateTime.UtcNow.ToString("yyyyMMddhhmmss")}",
```

Comment:
Maybe some sort of "strategy" and let the caller decide how to name the branch?

I used time stamps originally because I needed a unique branch name.